hamilton/graph_types.py

```diff
@@ -154,9 +154,13 @@ def version(self) -> typing.Optional[str]:
         The option `strip=True` means docstring and comments are ignored
         when hashing the function.
         """
-        if not self.originating_functions:
-            return None
+        if self.originating_functions is None or len(self.originating_functions) == 0:
```

Comment:
If `self.originating_functions == None`, `len(self.originating_functions)` should throw an exception "can't call len() on NoneType", no?

Reply:
Because it's an `OR` statement, it won't evaluate the second clause unless the first one is False.

Reply:
```python
if True or obj.asdfs.asf():
    print("passed")
# passed

if False or obj.asdfs.asf():
    print("not hit")
    
Traceback (most recent call last):
  File ... in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-5-7945ebca460e>", line 1, in <module>
    if False or obj.asdfs.asf():
NameError: name 'obj' is not defined
```